packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -88,7 +88,13 @@ module.exports = ({ strapi }) => ({
 
     const entities = await strapi.entityService.findMany(uid, params);
 
-    const mappedResults = await mapAsync(entities.results, (entity) => this.mapEntity(entity, uid));
+    if (!entities) {
+      return entities;
+    }
+
+    const mappedResults = await mapAsync(entities.results || [entities], (entity) =>
+      this.mapEntity(entity, uid)
+    );
     return { ...entities, results: mappedResults };
```

Comment:
what about doing this:
```suggestion
    if (entities.results) {
      const mappedResults = await mapAsync(entities.results, (entity) =>
        this.mapEntity(entity, uid)
      );
      return { ...entities, results: mappedResults };
    }

    // if entity is single type
    return this.mapEntity(entities, uid);
```

wdyt?

Reply:
oh nop, I was completely wrong, the frontend do expect that format :P